sdk/v2/core/secrets.go

```diff
@@ -0,0 +1,170 @@
+package core
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/brigadecore/brigade/sdk/v2/internal/restmachinery"
+	"github.com/brigadecore/brigade/sdk/v2/meta"
+)
+
+// SecretList is an ordered and pageable list of Secrets.
+type SecretList struct {
+	// ListMeta contains list metadata.
+	meta.ListMeta `json:"metadata"`
+	// Items is a slice of Secrets.
+	Items []Secret `json:"items,omitempty"`
+}
+
+// MarshalJSON amends SecretList instances with type metadata so that clients do
+// not need to be concerned with the tedium of doing so.
+func (s SecretList) MarshalJSON() ([]byte, error) {
+	type Alias SecretList
+	return json.Marshal(
+		struct {
+			meta.TypeMeta `json:",inline"`
+			Alias         `json:",inline"`
+		}{
+			TypeMeta: meta.TypeMeta{
+				APIVersion: meta.APIVersion,
+				Kind:       "SecretList",
+			},
+			Alias: (Alias)(s),
+		},
+	)
+}
+
+// Secret represents Project-level sensitive information.
+type Secret struct {
+	// Key is a key by which the secret can referred.
+	Key string `json:"key,omitempty"`
```

Comment:
Wondering if both the `Key` and `Value` fields should be deemed required via removing the `omitempty`s?

Reply:
`omitempty` doesn't have any real bearing on whether something is required or not. It really only affects the marshaled representation.

The question of validity is a server-side concern. The prototype API server would reject any secret that was lacking a value for either of those fields.

As for whether these fields should be marked "required" in the docs, I'm hesitant, because a secret, as with most other resources, can be sent in both directions. You can send a secret TO the API Server or retrieve one FROM it. When retrieving, it's perfectly plausible that the `Value` field would be empty because the prototype API server redacts the values on retrieval (and I expect our official API server will function the same way).

Reply:
Thanks for the explanation, especially helpful to remember that this object may take different forms depending on transmission type.